ngraph/core/src/node.cpp

```diff
@@ -155,12 +155,13 @@ void Node::set_arguments(const NodeVector& arguments) {
 }
 
 void Node::set_arguments(const OutputVector& arguments) {
+    // Remove existing inputs of this node
+    m_inputs.clear();
+
     // Add this node as a user of each argument.
     size_t i = 0;
     for (auto& output : arguments) {
-        auto output_node = output.get_node();
-        auto& output_descriptor = output_node->m_outputs.at(output.get_index());
-        m_inputs.emplace_back(this, i++, output_descriptor);
+        set_argument(i++, output);
```

Comment:
Good change! But can we add new C++ tests to cover the issue?

Reply:
Done.